frontend/javascripts/oxalis/model/reducers/reducer_helpers.js

```diff
@@ -1,25 +1,70 @@
 // @flow
 import Maybe from "data.maybe";
 
-import type { APIAnnotation, ServerBoundingBox } from "admin/api_flow_types";
-import type { Annotation, BoundingBoxObject } from "oxalis/store";
+import type { APIAnnotation, ServerBoundingBox, ServerUserBoundingBox } from "admin/api_flow_types";
+import type {
+  Annotation,
+  BoundingBoxObject,
+  UserBoundingBox,
+  UserBoundingBoxToServer,
+} from "oxalis/store";
 import type { Boundary } from "oxalis/model/accessors/dataset_accessor";
 import type { BoundingBoxType } from "oxalis/constants";
 import { V3 } from "libs/mjs";
 import * as Utils from "libs/utils";
 
+function convertServerBoundingBoxToMinMaxBoundingBox(
+  boundingBox: ServerBoundingBox,
+): BoundingBoxType {
+  return Utils.computeBoundingBoxFromArray(
+    Utils.concatVector3(Utils.point3ToVector3(boundingBox.topLeft), [
+      boundingBox.width,
+      boundingBox.height,
+      boundingBox.depth,
+    ]),
+  );
+}
+
 export function convertServerBoundingBoxToFrontend(
   boundingBox: ?ServerBoundingBox,
 ): ?BoundingBoxType {
   return Maybe.fromNullable(boundingBox)
-    .map(bb =>
-      Utils.computeBoundingBoxFromArray(
-        Utils.concatVector3(Utils.point3ToVector3(bb.topLeft), [bb.width, bb.height, bb.depth]),
-      ),
-    )
+    .map(bb => convertServerBoundingBoxToMinMaxBoundingBox(bb))
     .getOrElse(null);
 }
 
+export function convertUserBoundingBoxesFromServerToFrontend(
+  boundingBoxes: Array<ServerUserBoundingBox>,
+): Array<UserBoundingBox> {
+  return boundingBoxes.map(bb => {
+    const { color, id, name, isVisible, boundingBox } = bb;
+    const convertedBoundingBox = convertServerBoundingBoxToMinMaxBoundingBox(boundingBox);
+    return {
+      boundingBox: convertedBoundingBox,
+      color: color ? Utils.colorObjectToRGBArray(color) : Utils.getRandomColor(),
+      id,
+      name: name || `user bounding box ${id}`,
+      isVisible: isVisible != null ? isVisible : true,
+    };
+  });
+}
+
+export function convertUserBoundingBoxesFromFrontendToServer(
+  boundingBoxes: Array<UserBoundingBox>,
+): Array<UserBoundingBoxToServer> {
+  // The exact spreading is needed for flow to grasp that the conversion is correct.
+  return boundingBoxes.map(bb => {
+    const { id, boundingBox, name, color, isVisible } = bb;
+    return {
+      id,
+      name,
+      color,
+      isVisible,
+      boundingBox: Utils.computeBoundingBoxObjectFromBoundingBoxType(boundingBox),
+    };
```

Comment:
If you only modify a single property you can write this more concise :)

```suggestion
    const { boundingBox, ...rest } = bb;
    return { ...rest, boundingBox: Utils.computeBoundingBoxObjectFromBoundingBoxType(boundingBox) };
```

Reply:
That's genius :exploding_head: